### PR TITLE
Loading when data canceling booking

### LINKED ISSUE #79: Loading data when canceling booking
- Fix duplicate id radio button error
- Cancel booking without reloading data
- Update profile information in redux, storage, server
- Fomat price from USD to VND

### PR DESCRIPTION
**Pull request by thanhdo29**
## Related issues
Resolve #79 

## Summary

- Fix duplicate id radio button error
- Cancel booking without reloading data
- Update profile information in redux, storage, server
- Fomat price from USD to VND

## How to test
- Cancel booking without reloading data

1.  In the BookingUpcoming screen to cancel the booking
2.  In the BookingCancelled to check

- Update profile information in redux, storage, server

1. Perform profile update on the ProfileUserTemplate screen
2. Return to the ProfileTemplate check screen

### CODE REVIEW

**Comment by NguyenVanViet2004** [comment]
image

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
image

I chose the online payment option, but when I checked the appointment ticket, the payment method showed as cash?

**Comment by NguyenVanViet2004** [comment]
Please reformat the currency to VND
Simulator Screenshot - iPhone 16 Plus - 2024-12-11 at 15 03 24

**Comment by thanhdo29** [comment]
Here I check if the user selects but has not yet transferred money, I still show the radiobutton

**Comment by NguyenVanViet2004** [comment]
image
image
image

File: components/molecules/BookingUpcoming.tsx [hunk 1]
```diff
@@ -20,19 +21,31 @@ const BookingUpcoming = (): React.ReactElement => {
...
+    }, [refetch])
+  )
+
   const pendingAppointments = appointments.filter(
     (item) =>
       item.status === Status.PENDING &&
       !isNil(item.customer) &&
       item.customer.id === user.id
-  ).reverse()
+  )
```
Comment (NguyenVanViet2004) [inline components/molecules/BookingUpcoming.tsx:48]: `.reverse()`

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Replace this code block 
```html
<typeStyles.icon
      size={40}
      color={colors.white}
      alignSelf="center"
      padding={25}
      borderRadius={50}
      backgroundColor={typeStyles.color}
/>
```
with
```html
<View
    alignSelf='center'
    borderRadius={50}
    backgroundColor={typeStyles.color}>
    <typeStyles.icon size={50} color={colors.white} />
</View>
```

in [AppModal](https://github.com/NguyenVanViet2004/FE-GlowUp/blob/main/components/molecules/common/AppModal.tsx) to fix this bug (only occurs on ios device):
image

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
> [!IMPORTANT]
> CRASH APP 

Simulator Screen Recording - iPhone 16 Plus - 2024-12-13 at 11 38 13

**Comment by NguyenVanViet2004** [comment]
image

**Comment by NguyenVanViet2004** [comment]
| --- |
| --- |
|image||







https://github.com/user-attachments/assets/33896cb8-f8ac-4010-83a9-79d461d72b63


https://github.com/user-attachments/assets/7db5e426-1e6c-40fd-9664-20d9d000920f

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Fix

**Review by truongtdph40209 — APPROVED** [review]

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**